packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -0,0 +1,450 @@
+   * span when the response is finished.
+   * @param original The original patched function.
+   * @param options The arguments to the original function.
+   */
+  private getMakeRequestTraceFunction(
+    request: ClientRequest,
+    options: RequestOptions,
+    plugin: HttpPlugin
+  ): Func<ClientRequest> {
+    return (span: Span): ClientRequest => {
+      plugin._logger.debug('makeRequestTrace');
+
+      if (!span) {
+        plugin._logger.debug('makeRequestTrace span is null');
+        return request;
+      }
+
+      const propagation = plugin._tracer.getHttpTextFormat();
+      // If outgoing request headers contain the "Expect" header, the returned
+      // ClientRequest will throw an error if any new headers are added.
+      // So we need to clone the options object to be able to inject new
+      // header.
+      if (HttpPlugin.hasExpectHeader(options)) {
+        options = Object.assign({}, options);
+        options.headers = Object.assign({}, options.headers);
+      }
+      propagation.inject(span.context(), HttpPlugin.PROPAGATION_FORMAT, options.headers);
+
+      request.on('response', (response: IncomingMessage) => {
+        plugin._tracer.wrapEmitter(response);
+        plugin._logger.debug('outgoingRequest on response()');
+        response.on('end', () => {
+          plugin._logger.debug('outgoingRequest on end()');
+          const method = response.method ? response.method.toUpperCase() : 'GET';
+          const headers = options.headers;
+          const userAgent = headers ? headers['user-agent'] || headers['User-Agent'] : null;
+
+          const host = options.hostname || options.host || 'localhost';
+          span
+            .setAttribute(HttpPlugin.ATTRIBUTE_HTTP_HOST, host)
+            .setAttribute(HttpPlugin.ATTRIBUTE_HTTP_METHOD, method);
+          if (options.path) {
+            span
+              .setAttribute(HttpPlugin.ATTRIBUTE_HTTP_PATH, options.path)
+              .setAttribute(HttpPlugin.ATTRIBUTE_HTTP_ROUTE, options.path);
+          }
+
+          if (userAgent) {
+            span.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_USER_AGENT, userAgent.toString());
+          }
+          if (response.statusCode) {
+            span.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_STATUS_CODE, response.statusCode.toString());
+            span.setStatus(HttpPlugin.parseResponseStatus(response.statusCode));
+          }
+
+          if (plugin.options.applyCustomAttributesOnSpan) {
+            plugin.options.applyCustomAttributesOnSpan(span, request, response);
+          }
+
+          span.end();
+        });
+        HttpPlugin.setSpanOnError(span, response);
+      });
+
+      HttpPlugin.setSpanOnError(span, request);
+
+      plugin._logger.debug('makeRequestTrace return request');
+      return request;
+    };
+  }
+
+  private incomingRequestFunction(original: (event: string) => boolean, plugin: HttpPlugin) {
+    return function incomingRequest(event: string, ...args: unknown[]): boolean {
```

Comment:
try adding `this: {}` as first argument and remove the @ts-ignore (it will be removed in the output)

Reply:
Thanks for the tips